### PR TITLE
fix: optimize DB pull

### PR DESCRIPTION
**Pull request by pengx17**
When SQLite DB is pulled, its updates will be applied to a local yDoc. This is a blocking operation and surprisingly will also block the renderer process. See https://twitter.com/NumaanAshraf/status/968496732278374400

This PR proposed a fix that it will offload merge updates in a worker thread and then only apply a single update in the main process, without blocking the UI for too long.

This shall be a temporary solution. We may think about moving all DB operations off main process later.

Also added some tests for this fix

### CODE REVIEW

File: apps/electron/layers/main/src/workers/merge-update.worker.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { parentPort, workerData } from 'node:worker_threads';
+
+import * as Y from 'yjs';
+
+function getMergeUpdate(updates: Uint8Array[]) {
```
Comment (Brooooooklyn) [inline apps/electron/layers/main/src/workers/merge-update.worker.ts:5]: Does `SharedArrayBuffer` work here?

Comment (pengx17) [reply apps/electron/layers/main/src/workers/merge-update.worker.ts:5]: Should do, let me have a try

File: apps/electron/layers/main/src/workers/index.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      const buffer = new SharedArrayBuffer(update.byteLength);
+      const view = new Uint8Array(buffer);
+      view.set(update);
+      return view;
+    });
+
+    const worker = new Worker(workerFile, {
+      workerData: sharedArrayBufferUpdates,
+    });
+
```
Comment (himself65) [inline apps/electron/layers/main/src/workers/index.ts:26]: use async-call-rpc with the worker. I will handle this part

**Review by Himself65 — CHANGES_REQUESTED** [review]

File: apps/electron/layers/main/src/workers/index.ts (hunk 2 above)
Comment (pengx17) [reply apps/electron/layers/main/src/workers/index.ts:26]: Will you work on it in this PR or perhaps in a following one?

**Review by Himself65 — APPROVED** [review]

**Pull request merged**